File: app/src/main/java/com/spacex/launches/spacexrocketlaunches/AppComponent.java
package com.spacex.launches.spacexrocketlaunches;

import android.app.Application;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;
import dagger.android.support.AndroidSupportInjectionModule;
import javax.inject.Singleton;

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, ActivityBuilder.class, AppModule.class})
public interface AppComponent extends AndroidInjector<DaggerApplication> {

  void inject(RocketLaunchesApp crowdScoresApplication);

  @Override
  void inject(DaggerApplication instance);

  @Component.Builder
  interface Builder {
    @BindsInstance
    Builder application(Application application);

    AppComponent build();
  }
}

File: app/src/main/kotlin/com/spacex/ui/di/AppComponent.kt
package com.spacex.ui.di

import android.app.Application
import com.spacex.ui.base.RocketLaunchesApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, ActivityBuilder::class, AppModule::class])
interface AppComponent : AndroidInjector<DaggerApplication> {

    fun inject(rocketLaunchesApp: RocketLaunchesApp)

    override fun inject(instance: DaggerApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}
